sympy/core/numbers.py

```diff
@@ -1872,7 +1873,7 @@ def __floor__(self):
 
     def __ceil__(self):
         return self.ceiling()
-
+#ping
     def __eq__(self, other):
         try:
             other = _sympify(other)
```

Comment:
Similar: only convert when appropriate, use `NotImplemented` and possibly use `other.is_Number` instead of `isinstance(other, Number)